accepted/00062-hub-xmlrpc-api.md

```diff
@@ -0,0 +1,129 @@
+       - If they prefer to manage Clients via users that have no permissions to manage Servers, they can create a Hub-Org with no Servers. Any Hub-User in this Hub-Org will be able to connect to the Gateway, and possibly have permissions on Clients (if Servers have Server-Users with same name and passwords)
+
+* Multicasting: call an XMLRPC method on _N_ Servers at once
+  * method name stays the same
+  * Assuming the method on the Server accepts _M_ parameters, the same method on the Hub will accept an array of _N_ parameters, each of which contains as elements the _M_ parameters for a given Server
+    * eg. `actionchain.listChains(sessionKey)` → `hub.actionchain.listChains([serverSessionKey1, serverSessionKey2, ...])`
+    * eg. `channel.org.enableAccess(sessionKey, channelLabel, orgId)` → `hub.channel.org.enableAccess([serverSessionKey1, channelLabel1, orgId1], [serverSessionKey2, channelLabel2, orgId2], ...)`
+  * return types would be arrays of _N_ elements
+    * special elements would be needed to signal Exceptions
+
+* Availability
+  * if a Server is down at the time a Gateway call targeting it is made, call should fail after a configurable timeout. If multiple Servers are targeted, only that call fails and others continue
+
+
+## Impact on existing components and users
+
+All code would be new in a new component, so no change to existing components is expected. No impact on existing users is expected at all.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+We are not adding any Hub-centered functionality basically, this is a little more than a router.
+
+## Limitations
+
+- implementation of API methods currently not implemented on the Server, eg. downstream [#2361](https://github.com/SUSE/spacewalk/issues/2361) [#2362](https://github.com/SUSE/spacewalk/issues/2362), is explicitly not covered by this RFC
+
+
+# Alternatives
+[alternatives]: #alternatives
+* Authorization and authentication
+  * Alternative idea 1: a configuration file on the Hub contains a list of credentials and `serverIds`. Hub would automatically maintain a pool of Server `sessionKey`s from those credentials
+    * `hub.login(username, password)` → `hubSessionKey`
+    * `hub.serverMethod(hubSessionKey, serverId, parameters)` → `output`
+  * Alternative idea 2: god mode. We would assume Hub is root on Servers, so somehow automatically gets full administrative API access to all anyway
+
+* Multicasting
+* Alternative idea 2: each of the _M_ parameters is replaced by an array of _N_ elements
+  * eg. `actionchain.listChains(sessionKey)` → `hub.actionchain.listChains([serverSessionKey1, serverSessionKey2, ...])`
+  * eg. `channel.org.enableAccess(sessionKey, channelLabel, orgId)` → `hub.channel.org.enableAccess([serverSessionKey1, serverSessionKey2, ...], [channelLabel1, channelLabel2, ...], [orgId1, orgId2, ...])`
+  * return types would be arrays of _N_ elements
```

Comment:
Potential risk with that: What if the parameters have changed and one of the servers is older/newer than the others? Very unlikely though. In general, I like the non-alternative approach better.

Reply:
Agreed, that I also clarified in limitations anyway.